edb/server/args.py

```diff
@@ -1222,24 +1230,33 @@ def parse_args(**kwargs: Any):
                     method = ServerAuthMethod.Trust
                 else:
                     method = ServerAuthMethod.mTLS
+                methods[pos] = method
             else:
-                method = DEFAULT_AUTH_METHODS.get(transport)
-            methods[transport] = method
+                methods = (
+                    methods[:pos]
+                    + DEFAULT_AUTH_METHODS.get(transport)
+                    + methods[pos + 1:]
+                )
+            transport_methods[transport] = [method]
```

Comment:
Should it not be `= methods`?

Reply:
Yeah, that logic was wrong entirely, I pushed a fix.